tests/testthat/test-hive.R

```diff
@@ -0,0 +1,172 @@
+context("Hive translation")
+
+# Initialize simulated DB connection -------------------------------------
+
+hive_connection <- dbplyr:::simulate_hive()
+
+# Hive base_scalar conversions -----------------------------------------
+
+test_that("as.numeric field coercion translates to Hive's DOUBLE ", {
```

Comment:
Given that the underlying code is 90% the same as the other odbc backends, I think you only need a couple of tests for the things that are different.